### ISSUE TITLE
deleteSMS doesn't work

### ISSUE THREAD

**Issue by mmikel** [floatinghotpot/cordova-plugin-sms#4]
SMS.deleteSMS({box: '', address: gate_address}, function(n) {
  alert(n + ' sms messages deleted');
}, function(err){
  alert('error delete sms: ' + err);
});

showing "X sms messages deleted" and nothing ((

Android 4.4.2

**Comment by floatinghotpot** [comment]
What is X in the message? And address filter should be the sent phone number, what is gate_address here?

**Comment by mmikel** [comment]
Ok...

    var gate_address = '+79050850161';

        CleanupSMS: function () {
            alert(gate_address);
            SMS.deleteSMS({box: '', address: gate_address}, function(n) {
                alert(n + ' sms messages deleted');
            }, function(err){
                alert('error delete sms: ' + err);
            });
        },

first screenshot: gate_address is 79050850161
![screenshot_2015-02-17-06-43-10](https://cloud.githubusercontent.com/assets/11029191/6223067/11cc78fa-b671-11e4-85a6-7bf8d009ea61.png)

second screenshot: no error, 5 messages should be deleted
![screenshot_2015-02-17-06-43-17](https://cloud.githubusercontent.com/assets/11029191/6223069/11cef97c-b671-11e4-93e1-75db837d9969.png)

third screenshot: all messages at their places ((
![screenshot_2015-02-17-06-43-35](https://cloud.githubusercontent.com/assets/11029191/6223068/11ccc01c-b671-11e4-921c-98e566275ea6.png)

**Comment by floatinghotpot** [comment]
I tested all APIs on my samsung s4 (android 4.3) and it worked pretty well. 

but not tested on android 4.4 device yet.

**Comment by mmikel** [comment]
looks like kitkat has some changes in sms

http://android-developers.blogspot.ru/2013/10/getting-your-sms-apps-ready-for-kitkat.html

**Comment by floatinghotpot** [comment]
Yeah, you are right. I know that access to SMS has been restricted since 4.4.

I remember that it's not allowed to intercept SMS any more, but not sure that deleting is not allowed either...

**Comment by firaskudsy** [comment]
@floatinghotpot 
i tried this pluging on android 4.1.2 and the delete failed !!! any help ?? the listSMS worked fine that mean the plugin is ok , but the delete do nothing.

  $scope.delete = function(id) {

console.log("id = " + id);
            if(SMS) SMS.deleteSMS({
                _id : id
            }, function( n ){
                console.log( n + ' sms messages deleted' );
            }, function(err){
                console.log('error delete sms: ' + err);
            });

any help ?

**Comment by trippo** [comment]
I obtain the same error on Android 4.1.2

**Comment by trippo** [comment]
On android 5 work without error but not delete nothing, the list work fine

**Comment by trippo** [comment]
i found a fix oyu must change

ctx.getContentResolver().delete(uri, "_id=" + id, (String[])null);

to

ctx.getContentResolver().delete(Uri.parse("content://sms/" + id),null, (String[])null);

**Comment by AlbanWende** [comment]
@trippo , did it work fine for you?

**Comment by trippo** [comment]
Yes on android 4.1.2

**Comment by floatinghotpot** [comment]
As to intercept / delete, it depends on the Android OS version. It’s quite okay on version before Android 4.4+. 

But after v4.4, as Google add very strict security control after v4.4, and only the system default SMS app can intercept / delete SMS. 

If your app is not the default SMS app, you can still get & read SMS, but cannot intercept / delete (remove SMS before other app get it) any more.

**Comment by prantikv** [comment]
Yes i +1 this delete and intercept is not working anymore on devices >4.4.
Still can read messages and setup alerts on sms arrival.

**Comment by Brianchat** [comment]
This still seems to be the case.

**Comment by mbhd90** [comment]
For me it not working? any solution?
